pkg/sources/kafka/metrics.go

```diff
@@ -23,27 +23,6 @@ import (
 	"github.com/numaproj/numaflow/pkg/metrics"
 )
 
-// kafkaSourceReadCount is used to indicate the number of messages read
-var kafkaSourceReadCount = promauto.NewCounterVec(prometheus.CounterOpts{
-	Subsystem: "kafka_source",
-	Name:      "read_total",
-	Help:      "Total number of messages Read",
-}, []string{metrics.LabelVertex, metrics.LabelPipeline, metrics.LabelPartitionName})
-
-// kafkaSourceAckCount is used to indicate the number of messages Acknowledged
-var kafkaSourceAckCount = promauto.NewCounterVec(prometheus.CounterOpts{
-	Subsystem: "kafka_source",
-	Name:      "ack_total",
-	Help:      "Total number of messages Acknowledged",
-}, []string{metrics.LabelVertex, metrics.LabelPipeline})
-
-// kafkaSourceOffsetAckErrors is used to indicate the number of errors while reading from kafka source with offsets
-var kafkaSourceOffsetAckErrors = promauto.NewCounterVec(prometheus.CounterOpts{
-	Subsystem: "kafka_source",
-	Name:      "ack_error_total",
-	Help:      "Total number of Kafka ID Errors",
-}, []string{metrics.LabelVertex, metrics.LabelPipeline})
-
 // kafkaPending is used to indicate the number of messages pending in the kafka source
 var kafkaPending = promauto.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
can we also remove this one?

Reply:
we have `vertex_pending_messages` but I'm doubtful whether it can replace `kafkaPending` and `udsourcePending`. as per my understanding, 

```
if pending, err := lagReader.Pending(ctx); err != nil {
					log.Errorw("Failed to get pending messages", zap.Error(err))
```
here we are calling below to build up pending info

```
func (ks *kafkaSource) Pending(_ context.Context) (int64, error) {
	if ks.adminClient == nil || ks.saramaClient == nil {
```

@whynowy can you please help here

Reply:
Let's keep it for now? In case ppl want to see the real time kafka pending.